src/generic/renderg.cpp

```diff
@@ -455,7 +455,7 @@ wxRendererGeneric::DrawHeaderButtonContents(wxWindow *win,
 
         wxDCFontChanger setFont(dc, font);
         wxDCTextColourChanger setTextFg(dc, clr);
-        wxDCTextBgModeChanger setBgMode(dc, wxBRUSHSTYLE_TRANSPARENT);
```

Comment:
This undoes new intentions mentioned in 3f3ea4d266. In this case both rendering changes aren't needed and the problem is with `wxDCTextBgModeChanger` which doesn't set `m_modeOld` to `wxBRUSHSTYLE_INVALID` in its second ctor, resulting in the attribute never being (re)stored.

Reply:
I've reverted the commit and fixed `wxDCTextBgModeChanger`.